### PR TITLE
Implement copy_, fill_, and ones_like for Nested Tensors backends

### PR DESCRIPTION
**Pull request by ani300**
Summary: This diff implements copy_ in order to allow pinned memory transfers for nested tensors, as well as fill_ and ones_like, to test whether nested tensors can be created with other factory functions.

Test Plan: Pass all CI and sandcastle jobs.

Reviewed By: mikekgfb

Differential Revision: D40689594

### CODE REVIEW

File: torch/nested/__init__.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def nested_tensor(tensor_list: List[Tensor], *, dtype: Optional[DType] = None, d
         >>> nt.is_leaf
         True
     """
-    if not isinstance(tensor_list, list) or any([not torch.is_tensor(t) for t in tensor_list]):
+    if not isinstance(tensor_list, list) or any([not isinstance(t, torch.Tensor) for t in tensor_list]):
```
Comment (cpuhrsch) [inline torch/nested/__init__.py:44]: Why this change?

File: test/test_nestedtensor.py [hunk 2]
```diff
@@ -1,18 +1,26 @@
 # Owner(s): ["module: nestedtensor"]
 
+import unittest
```
Comment (cpuhrsch) [inline test/test_nestedtensor.py:3]: Can we add unit tests for these ops?

File: torch/nested/__init__.py (hunk 1 above)
Comment (ani300) [reply torch/nested/__init__.py:44]: I think it was done automatically from a linter suggestion, I can revert it

File: test/test_nestedtensor.py (hunk 2 above)
Comment (ani300) [reply test/test_nestedtensor.py:3]: I'll add them

File: aten/src/ATen/native/nested/NestedTensorFactories.cpp [hunk 3]
```diff
@@ -106,9 +106,26 @@ Tensor _to_copy_nested(
...
+
+Tensor& fill_nested_(Tensor& self, const Scalar& value) {
+  const auto& self_buf = get_nested_tensor_impl(self)->get_buffer();
+  self_buf.fill_(value);
+  return self;
+}
+
+Tensor& fill_nested_(Tensor& self, const Tensor& value) {
+  const auto& self_buf = get_nested_tensor_impl(self)->get_buffer();
+  self_buf.fill_(value);
```
Comment (cpuhrsch) [inline aten/src/ATen/native/nested/NestedTensorFactories.cpp:126]: This won't work if value is anything but a scalar. In the test you only all it with scalar. Do you need to support this overload for your use case?

Comment (ani300) [reply aten/src/ATen/native/nested/NestedTensorFactories.cpp:126]: No. I can add the test that fails with a non-0-dimensional tensor too. I think the reason this exists is so that you can do something like t.fill_(other_t[0]), which returns a 0-dimensional tensor instead of a scalar.

**Review by cpuhrsch — APPROVED** [review]
